humility-core/src/reflect.rs

```diff
@@ -599,13 +599,33 @@ impl std::ops::Deref for Array {
     }
 }
 
+impl Array {
+    fn format_as_c_string(&self, out: &mut dyn std::io::Write) -> Result<()> {
+        let mut bytes = Vec::new();
+        for v in &self.0 {
+            let b = v.as_base().ok().and_then(Base::as_u8).context(
+                "cannot interpret array of non-u8 values as a C string",
+            )?;
+            if b == 0 {
+                break;
+            }
+            bytes.push(b);
+        }
```

Comment:
You _could_ accumulate a `Result<Vec<u8>, _>` and make this immutable, e.g.
```suggestion
        let bytes = self
            .0
            .iter()
            .map(|b| {
                b.as_base().ok().and_then(Base::as_u8).context(
                    "cannot interpret array of non-u8 values as a C string",
                )
            })
            .take_while(|b| b.as_ref().map(|b| *b != 0).unwrap_or(true))
            .collect::<Result<Vec<u8>, _>>()?;
```

Not sure if you want to, though!

Reply:
I don't think I do, hah. The `.take_while()` line seems overly dense for the operation it's doing ("stop at the first 0").